File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.py
def fecto(n):
    f = 1 
    for i in range(1, n+1):
        f *= i
    return f

def solution(balls, share):

    answer = fecto(balls) / (fecto(balls - share) * fecto(share))
    return answer

print(solution(3 , 2))
print(solution(5, 3))

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
class Solution {
    public long solution(int balls, int share) {
        long answer = 0;

        int d = (balls - share) > share ? share : balls - share;
        if (d == 0) return 1;

        return solution(balls - 1, d - 1) * balls / d;
    }

}